<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>bioScience quickstart &mdash; bioscience 0.0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Load dataset" href="../userGuide/load.html" />
    <link rel="prev" title="Installation" href="install.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            bioscience
          </a>
              <div class="version">
                0.0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="whatis.html">What is bioScience?</a></li>
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">bioScience quickstart</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#run-on-sequential-mode-cpu">Run on sequential mode (CPU)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#run-on-parallel-mode-cpu">Run on parallel mode (CPU)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#run-on-parallel-mode-gpu">Run on parallel mode (GPU)</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../userGuide/load.html">Load dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../userGuide/preprocess.html">Pre-processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../userGuide/dataMining.html">Data mining</a></li>
<li class="toctree-l1"><a class="reference internal" href="../userGuide/results.html">Results</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../api/api.html">API reference</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contributor's Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributors/contributing.html">Contributing to bioScience</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributors/developmentProcess.html">Development Process</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributors/about.html">About us</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributors/release.html">Release History</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">bioscience</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">bioScience quickstart</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/gettingStarted/quickstart.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="bioscience-quickstart">
<h1>bioScience quickstart<a class="headerlink" href="#bioscience-quickstart" title="Permalink to this headline"></a></h1>
<p><strong>All implemented modes</strong> are associated with examples, check
<a class="reference external" href="https://github.com/aureliolfdez/bioscience/tree/main/tests/test_integration">“bioScience examples”</a>
for more information.</p>
<hr class="docutils" />
<section id="run-on-sequential-mode-cpu">
<h2>Run on sequential mode (CPU)<a class="headerlink" href="#run-on-sequential-mode-cpu" title="Permalink to this headline"></a></h2>
<p><a class="reference external" href="https://github.com/aureliolfdez/bioscience/tree/main/tests/test_integration/test_sequential.py">“tests/test_integration/test_sequential.py”</a>
demonstrates basic API functionality to generate results from the application of a data mining technique for gene expression datasets. This source code is executed sequentially on the CPU processors.</p>
<ol class="arabic">
<li><p><strong>Load gene co-expression dataset:</strong> Load the gene co-expression dataset from the input file. The first option loads a binary dataset, while the second option loads a non-binary gene co-expression dataset. Finally, it is shown how to load an RNA-Seq dataset.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">bioscience</span> <span class="k">as</span> <span class="nn">bs</span>

<span class="c1"># Binary dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/binaryTest3.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">naFilter</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>

<span class="c1"># Non-binary dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/synthetic3.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">naFilter</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>

<span class="c1"># RNA-Seq dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/rnaseq.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">index_lengths</span><span class="o">=</span><span class="mi">1</span> <span class="p">,</span><span class="n">naFilter</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Preprocessing:</strong> In this phase, different methods for pre-processing are shown. They range from basic preprocessing methods such as discretisation, standardisation and normalisation among others to preprocessing methods used on RNA-Seq data. Finally, bioScience also has methods to binarise a gene expression dataset.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># 2.1) Standard preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">discretize</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">n_bins</span><span class="o">=</span> <span class="mi">2</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">standardize</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">scale</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">normalDistributionQuantile</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>

<span class="c1"># 2.2) RNA-Seq preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">tpm</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">cpm</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>

<span class="c1"># 2.3) Binary preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">binarize</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">listDatasets</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">binarizeLevels</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">inactiveLevel</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">activeLevel</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span> <span class="n">soc</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
<span class="n">listDatasets</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">binarizeLevels</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Data mining:</strong> This third phase is responsible for executing the data mining techniques of the user’s choice. In this case, the example shows the execution of a Binary Biclustering algorithm called BiBit. In addition, by means of the option <code class="docutils literal notranslate"><span class="pre">mode=1</span></code>, it is being specified that this algorithm is to be executed sequentially.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># BiBit algorithm</span>
<span class="c1"># Single dataset</span>
<span class="n">listModels</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">bibit</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">cMnr</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cMnc</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">mode</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">debug</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>

<span class="c1"># List of datasets (if bs.binarizeLevels function is used)</span>
<span class="n">listModels</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">bibit</span><span class="p">(</span><span class="n">listDatasets</span><span class="p">,</span> <span class="n">cMnr</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cMnc</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">mode</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">debug</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Results:</strong> Save gene names for each result generated by the data mining technique.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">bs</span><span class="o">.</span><span class="n">saveGenes</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;/path/&quot;</span><span class="p">,</span> <span class="n">models</span><span class="o">=</span><span class="n">listModels</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">dataset</span><span class="p">)</span>
</pre></div>
</div>
</li>
</ol>
</section>
<section id="run-on-parallel-mode-cpu">
<h2>Run on parallel mode (CPU)<a class="headerlink" href="#run-on-parallel-mode-cpu" title="Permalink to this headline"></a></h2>
<p><a class="reference external" href="https://github.com/aureliolfdez/bioscience/tree/main/tests/test_integration/test_parallel_cpu.py">“tests/test_integration/test_parallel_cpu.py”</a>
demonstrates basic API functionality to generate results from the application of a data mining technique for gene expression datasets. This source code is executed in a parallel environment on CPU processors.</p>
<ol class="arabic">
<li><p><strong>Load gene co-expression dataset:</strong> Load the gene co-expression dataset from the input file. The first option loads a binary dataset, while the second option loads a non-binary gene co-expression dataset. Finally, it is shown how to load an RNA-Seq dataset.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">bioscience</span> <span class="k">as</span> <span class="nn">bs</span>

<span class="c1"># Binary dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/binaryTest3.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">naFilter</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>

<span class="c1"># Non-binary dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/synthetic3.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">naFilter</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>

<span class="c1"># RNA-Seq dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/rnaseq.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">index_lengths</span><span class="o">=</span><span class="mi">1</span> <span class="p">,</span><span class="n">naFilter</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Preprocessing:</strong> In this phase, different methods for pre-processing are shown. They range from basic preprocessing methods such as discretisation, standardisation and normalisation among others to preprocessing methods used on RNA-Seq data. Finally, bioScience also has methods to binarise a gene expression dataset.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># 2.1) Standard preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">discretize</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">n_bins</span><span class="o">=</span> <span class="mi">2</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">standardize</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">scale</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">normalDistributionQuantile</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>

<span class="c1"># 2.2) RNA-Seq preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">tpm</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">cpm</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>

<span class="c1"># 2.3) Binary preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">binarize</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">listDatasets</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">binarizeLevels</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">inactiveLevel</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">activeLevel</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span> <span class="n">soc</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
<span class="n">listDatasets</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">binarizeLevels</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Data mining:</strong> This third phase is responsible for executing the data mining techniques of the user’s choice. In this case, the example shows the execution of a Binary Biclustering algorithm called BiBit. In addition, by means of the option <code class="docutils literal notranslate"><span class="pre">mode=2</span></code>, it is being specified that this algorithm is to be executed sequentially.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># BiBit algorithm</span>
<span class="c1"># Single dataset</span>
<span class="n">listModels</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">bibit</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">cMnr</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cMnc</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">mode</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">debug</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>

<span class="c1"># List of datasets (if bs.binarizeLevels function is used)</span>
<span class="n">listModels</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">bibit</span><span class="p">(</span><span class="n">listDatasets</span><span class="p">,</span> <span class="n">cMnr</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cMnc</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">mode</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">debug</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Results:</strong> Save gene names for each result generated by the data mining technique.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">bs</span><span class="o">.</span><span class="n">saveGenes</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;/path/&quot;</span><span class="p">,</span> <span class="n">models</span><span class="o">=</span><span class="n">listModels</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">dataset</span><span class="p">)</span>
</pre></div>
</div>
</li>
</ol>
</section>
<section id="run-on-parallel-mode-gpu">
<h2>Run on parallel mode (GPU)<a class="headerlink" href="#run-on-parallel-mode-gpu" title="Permalink to this headline"></a></h2>
<p><a class="reference external" href="https://github.com/aureliolfdez/bioscience/tree/main/tests/test_integration/test_parallel_gpu.py">“tests/test_integration/test_parallel_gpu.py”</a>
demonstrates basic API functionality to generate results from the application of a data mining technique for gene expression datasets. This source code is executed in a parallel environment on GPU devices.</p>
<ol class="arabic">
<li><p><strong>Load gene co-expression dataset:</strong> Load the gene co-expression dataset from the input file. The first option loads a binary dataset, while the second option loads a non-binary gene co-expression dataset. Finally, it is shown how to load an RNA-Seq dataset.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">bioscience</span> <span class="k">as</span> <span class="nn">bs</span>

<span class="c1"># Binary dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/binaryTest3.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">naFilter</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>

<span class="c1"># Non-binary dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/synthetic3.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">naFilter</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>

<span class="c1"># RNA-Seq dataset load</span>
<span class="n">dataset</span> <span class="o">=</span> <span class="n">load</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;datasets/rnaseq.txt&quot;</span><span class="p">,</span> <span class="n">index_gene</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">index_lengths</span><span class="o">=</span><span class="mi">1</span> <span class="p">,</span><span class="n">naFilter</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">head</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Preprocessing:</strong> In this phase, different methods for pre-processing are shown. They range from basic preprocessing methods such as discretisation, standardisation and normalisation among others to preprocessing methods used on RNA-Seq data. Finally, bioScience also has methods to binarise a gene expression dataset.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># 2.1) Standard preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">discretize</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">n_bins</span><span class="o">=</span> <span class="mi">2</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">standardize</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">scale</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">normalDistributionQuantile</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>

<span class="c1"># 2.2) RNA-Seq preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">tpm</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">bs</span><span class="o">.</span><span class="n">cpm</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>

<span class="c1"># 2.3) Binary preprocessing</span>
<span class="n">bs</span><span class="o">.</span><span class="n">binarize</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
<span class="n">listDatasets</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">binarizeLevels</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">inactiveLevel</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span> <span class="n">activeLevel</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span> <span class="n">soc</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
<span class="n">listDatasets</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">binarizeLevels</span><span class="p">(</span><span class="n">dataset</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Data mining:</strong> This third phase is responsible for executing the data mining techniques of the user’s choice. In this case, the example shows the execution of a Binary Biclustering algorithm called BiBit. In addition, by means of the option <code class="docutils literal notranslate"><span class="pre">mode=3</span></code>, it is being specified that this algorithm is to be executed sequentially.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># BiBit algorithm</span>
<span class="c1"># Single dataset</span>
<span class="n">listModels</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">bibit</span><span class="p">(</span><span class="n">dataset</span><span class="p">,</span> <span class="n">cMnr</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cMnc</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">mode</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">deviceCount</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">debug</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>

<span class="c1"># List of datasets (if bs.binarizeLevels function is used)</span>
<span class="n">listModels</span> <span class="o">=</span> <span class="n">bs</span><span class="o">.</span><span class="n">bibit</span><span class="p">(</span><span class="n">listDatasets</span><span class="p">,</span> <span class="n">cMnr</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cMnc</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">mode</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">deviceCount</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">debug</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p><strong>Results:</strong> Save gene names for each result generated by the data mining technique.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">bs</span><span class="o">.</span><span class="n">saveGenes</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="s2">&quot;/path/&quot;</span><span class="p">,</span> <span class="n">models</span><span class="o">=</span><span class="n">listModels</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">dataset</span><span class="p">)</span>
</pre></div>
</div>
</li>
</ol>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="install.html" class="btn btn-neutral float-left" title="Installation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../userGuide/load.html" class="btn btn-neutral float-right" title="Load dataset" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, bioScience Core Development Team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>