<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>API reference &mdash; bioscience 0.0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Contributing to bioScience" href="../contributors/contributing.html" />
    <link rel="prev" title="Results" href="../userGuide/results.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            bioscience
          </a>
              <div class="version">
                0.0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../gettingStarted/whatis.html">What is bioScience?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../gettingStarted/install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../gettingStarted/quickstart.html">bioScience quickstart</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">User guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../userGuide/load.html">Load dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../userGuide/preprocess.html">Pre-processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../userGuide/dataMining.html">Data mining</a></li>
<li class="toctree-l1"><a class="reference internal" href="../userGuide/results.html">Results</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">API reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#module-bioscience.base.files">bioscience.base</a></li>
<li class="toctree-l2"><a class="reference internal" href="#module-bioscience.preprocess.Standard">bioscience.preprocess</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bioscience-datamining">bioscience.dataMining</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#module-bioscience.dataMining.biclustering.Biclustering">Biclustering</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contributor's Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributors/contributing.html">Contributing to bioScience</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributors/developmentProcess.html">Development Process</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributors/about.html">About us</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributors/release.html">Release History</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">bioscience</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">API reference</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/api/api.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="api-reference">
<h1>API reference<a class="headerlink" href="#api-reference" title="Permalink to this headline"></a></h1>
<p>This page gives an overview of all public bioScience objects, functions and methods. All classes and functions exposed in <code class="docutils literal notranslate"><span class="pre">bioscience.*</span></code> namespace are public.</p>
<p>The following subpackages are public.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">bioscience.base</span></code>: It contains functions for the management of I/O operations, files and objects and methods that form the core of the library and are used by the rest of the library’s subpackages.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">bioscience.preprocess</span></code>: This subpackage includes the functions associated with each of the preprocessing methods implemented in bioScience.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">bioscience.dataMining</span></code>: This subpackage contains all the source code for those data mining techniques that can be used in bioScience.</p></li>
</ul>
<p>Within each subpackage there may be various functions and methods that may be private because they are often intermediate operations of the implemented methods.</p>
<div class="toctree-wrapper compound">
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">API reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#module-bioscience.base.files">bioscience.base</a></li>
<li class="toctree-l2"><a class="reference internal" href="#module-bioscience.preprocess.Standard">bioscience.preprocess</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bioscience-datamining">bioscience.dataMining</a></li>
</ul>
</li>
</ul>
</div>
<section id="module-bioscience.base.files">
<span id="bioscience-base"></span><h2>bioscience.base<a class="headerlink" href="#module-bioscience.base.files" title="Permalink to this headline"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="bioscience.base.files.load">
<span class="sig-prename descclassname"><span class="pre">bioscience.base.files.</span></span><span class="sig-name descname"><span class="pre">load</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">separator</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'\t'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">skipr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">naFilter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">index_gene</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-</span> <span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">index_lengths</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-</span> <span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">head</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="headerlink" href="#bioscience.base.files.load" title="Permalink to this definition"></a></dt>
<dd><p>Load any data from a txt or csv file. (Reuse function)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>file_path</strong> (<em>str</em>) – The path where the file is stored.</p></li>
<li><p><strong>separator</strong> (<em>str</em><em>,</em><em>optional</em>) – An attribute indicating how the columns of the file are separated.</p></li>
<li><p><strong>skipr</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of rows the user wishes to omit from the file, defaults to 0.</p></li>
<li><p><strong>naFilter</strong> (<em>boolean</em><em>, </em><em>optional</em>) – Boolean to detect NA values in a file. NA values shall be replaced by 0’s, defaults to False.</p></li>
<li><p><strong>index_gene</strong> (<em>int</em><em>, </em><em>optional</em>) – Column position where the gene names are stored in the dataset, defaults to -1 (deactivated).</p></li>
<li><p><strong>index_lengths</strong> (<em>int</em><em>, </em><em>optional</em>) – Column position where the gene lengths are store in the dataset, defaults to -1 (deactivated).</p></li>
<li><p><strong>head</strong> (<em>int</em><em>, </em><em>optional</em>) – Row number(s) containing column labels and marking the start of the data (zero-indexed), defaults to None.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dataset object from the reading of a file</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.base.files.saveBinaryDatasets">
<span class="sig-prename descclassname"><span class="pre">bioscience.base.files.</span></span><span class="sig-name descname"><span class="pre">saveBinaryDatasets</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">datasets</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.files.saveBinaryDatasets" title="Permalink to this definition"></a></dt>
<dd><p>If the dataset has been binarised, this function allows storing the binary dataset.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>path</strong> (<em>str</em>) – The path where the file will be stored.</p></li>
<li><p><strong>datasets</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object which stores the binary dataset.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.base.files.saveGenes">
<span class="sig-prename descclassname"><span class="pre">bioscience.base.files.</span></span><span class="sig-name descname"><span class="pre">saveGenes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">models</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.files.saveGenes" title="Permalink to this definition"></a></dt>
<dd><p>Save the gene names from the results of applying a data mining technique.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>path</strong> (<em>str</em>) – The path where the file will be stored.</p></li>
<li><p><strong>models</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.dataMining.biclustering.BiclusteringModel</span></code>) – The results of the data mining technique.</p></li>
<li><p><strong>data</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object which stores the original dataset.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.base.files.saveResults">
<span class="sig-prename descclassname"><span class="pre">bioscience.base.files.</span></span><span class="sig-name descname"><span class="pre">saveResults</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">models</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.files.saveResults" title="Permalink to this definition"></a></dt>
<dd><p>Save the results of applying a data mining technique.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>path</strong> (<em>str</em>) – The path where the file will be stored.</p></li>
<li><p><strong>models</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.dataMining.biclustering.BiclusteringModel</span></code>) – The results of the data mining technique.</p></li>
<li><p><strong>data</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object which stores the original dataset.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.base.files.saveResultsIndex">
<span class="sig-prename descclassname"><span class="pre">bioscience.base.files.</span></span><span class="sig-name descname"><span class="pre">saveResultsIndex</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">models</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.files.saveResultsIndex" title="Permalink to this definition"></a></dt>
<dd><p>Save the results index (rows and columns index of the dataset) of applying a data mining technique.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>path</strong> (<em>str</em>) – The path where the file will be stored.</p></li>
<li><p><strong>models</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.dataMining.biclustering.BiclusteringModel</span></code>) – An attribute indicating how the columns of the file are separated.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<span class="target" id="module-bioscience.base.models"></span><dl class="py class">
<dt class="sig sig-object py" id="bioscience.base.models.Bicluster">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioscience.base.models.</span></span><span class="sig-name descname"><span class="pre">Bicluster</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">rows</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cols</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">validations</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.models.Bicluster" title="Permalink to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>This is a conceptual class representing a bicluster after applying a Biclustering technique.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>rows</strong> (<em>np.array</em>) – Rows of the bicluster.</p></li>
<li><p><strong>cols</strong> (<em>np.array</em><em>, </em><em>optional</em>) – Columns of the bicluster.</p></li>
<li><p><strong>data</strong> (<em>np.array</em><em>, </em><em>optional</em>) – Bicluster values according to the original dataset.</p></li>
<li><p><strong>validations</strong> (<em>np.array</em><em>, </em><em>optional</em>) – A set of instances from <a class="reference internal" href="#bioscience.base.models.Validation" title="bioscience.base.models.Validation"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Validation</span></code></a>.</p></li>
</ul>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Bicluster.cols">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">cols</span></span><a class="headerlink" href="#bioscience.base.models.Bicluster.cols" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the cols property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Bicluster.data">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">data</span></span><a class="headerlink" href="#bioscience.base.models.Bicluster.data" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the data property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Bicluster.rows">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">rows</span></span><a class="headerlink" href="#bioscience.base.models.Bicluster.rows" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the rows property.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bioscience.base.models.Bicluster.sizeBicluster">
<span class="sig-name descname"><span class="pre">sizeBicluster</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.models.Bicluster.sizeBicluster" title="Permalink to this definition"></a></dt>
<dd><p>Number of total elements in the bicluster.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="bioscience.base.models.Bicluster.sort">
<span class="sig-name descname"><span class="pre">sort</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.models.Bicluster.sort" title="Permalink to this definition"></a></dt>
<dd><p>Sort the column and row array by theirs indices.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Bicluster.validations">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">validations</span></span><a class="headerlink" href="#bioscience.base.models.Bicluster.validations" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the validations property.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="bioscience.base.models.BiclusteringModel">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioscience.base.models.</span></span><span class="sig-name descname"><span class="pre">BiclusteringModel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">results</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.models.BiclusteringModel" title="Permalink to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>This is a conceptual class representing a set of biclusters generated after applying a Biclustering technique.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>results</strong> (set(<a class="reference internal" href="#bioscience.base.models.Bicluster" title="bioscience.base.models.Bicluster"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Bicluster</span></code></a>), optional) – Data structure (set) that stores all biclusters after running a Biclustering algorithm.</p></li>
<li><p><strong>executionTime</strong> (<em>float</em><em>, </em><em>optional</em>) – Time taken to execute the Biclustering method.</p></li>
</ul>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.BiclusteringModel.executionTime">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">executionTime</span></span><a class="headerlink" href="#bioscience.base.models.BiclusteringModel.executionTime" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the executionTime property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.BiclusteringModel.results">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">results</span></span><a class="headerlink" href="#bioscience.base.models.BiclusteringModel.results" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the results property.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="bioscience.base.models.Dataset">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioscience.base.models.</span></span><span class="sig-name descname"><span class="pre">Dataset</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneNames</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">columnsNames</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lengths</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">annotations</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.models.Dataset" title="Permalink to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>This is a concept class representing a dataset.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>original</strong> (<em>np.array</em>) – Here the original dataset is stored in a NumPy array.</p></li>
<li><p><strong>data</strong> (<em>np.array</em>) – This attribute is used to store the dataset once it has undergone the transformations desired by the user.</p></li>
<li><p><strong>geneNames</strong> (<em>np.array</em><em>, </em><em>optional</em>) – Array with the name of the genes involved in the dataset. If the dataset does not have the name of the genes, it shall be replaced by a set of sequential numbers.</p></li>
<li><p><strong>columnsNames</strong> (<em>np.array</em><em>, </em><em>optional</em>) – Array with the name of the columns involved in the dataset. If the dataset does not have the name of the columns, it shall be replaced by a set of sequential numbers.</p></li>
<li><p><strong>lengths</strong> (<em>np.array</em><em>, </em><em>optional</em>) – Array with gene length value (RNA-Seq)</p></li>
<li><p><strong>annotations</strong> (<em>np.array</em><em>, </em><em>optional</em>) – Array that stores data from an annotation file for subsequent validation phases.</p></li>
<li><p><strong>cut</strong> (<em>float</em><em>, </em><em>optional</em>) – Cut-off parameter used in level binarisation.</p></li>
</ul>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Dataset.annotations">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">annotations</span></span><a class="headerlink" href="#bioscience.base.models.Dataset.annotations" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the annotations property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Dataset.columnsNames">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">columnsNames</span></span><a class="headerlink" href="#bioscience.base.models.Dataset.columnsNames" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the columnsNames property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Dataset.cut">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">cut</span></span><a class="headerlink" href="#bioscience.base.models.Dataset.cut" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the cut property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Dataset.data">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">data</span></span><a class="headerlink" href="#bioscience.base.models.Dataset.data" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the data property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Dataset.geneNames">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">geneNames</span></span><a class="headerlink" href="#bioscience.base.models.Dataset.geneNames" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the geneNames property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Dataset.lengths">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">lengths</span></span><a class="headerlink" href="#bioscience.base.models.Dataset.lengths" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the lengths property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Dataset.original">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">original</span></span><a class="headerlink" href="#bioscience.base.models.Dataset.original" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the original property.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="bioscience.base.models.Validation">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioscience.base.models.</span></span><span class="sig-name descname"><span class="pre">Validation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">measure</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">value</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.base.models.Validation" title="Permalink to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>This is a conceptual class representing a validation model after applying a data mining technique.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>measure</strong> (<em>str</em>) – Name of the validation measure used.</p></li>
<li><p><strong>value</strong> (<em>float</em>) – Value of the validation measure used.</p></li>
</ul>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Validation.measure">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">measure</span></span><a class="headerlink" href="#bioscience.base.models.Validation.measure" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the measure property.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="bioscience.base.models.Validation.value">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">value</span></span><a class="headerlink" href="#bioscience.base.models.Validation.value" title="Permalink to this definition"></a></dt>
<dd><p>Getter and setter methods of the value property.</p>
</dd></dl>

</dd></dl>

</section>
<section id="module-bioscience.preprocess.Standard">
<span id="bioscience-preprocess"></span><h2>bioscience.preprocess<a class="headerlink" href="#module-bioscience.preprocess.Standard" title="Permalink to this headline"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.Standard.discretize">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.Standard.</span></span><span class="sig-name descname"><span class="pre">discretize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_bins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">strategy</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'kmeans'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.Standard.discretize" title="Permalink to this definition"></a></dt>
<dd><p>Discretizes data into N bins.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be discretized.</p></li>
<li><p><strong>n_bins</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of bins to produce, defaults to 2.</p></li>
<li><p><strong>strategy</strong> (<em>str</em><em>, </em><em>optional</em>) – Strategy used to define the widths of the bins. Options (kmeans, quantile, uniforme). Defaults to ‘kmeans’.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.Standard.normalDistributionQuantile">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.Standard.</span></span><span class="sig-name descname"><span class="pre">normalDistributionQuantile</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">quantiles</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1000</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.Standard.normalDistributionQuantile" title="Permalink to this definition"></a></dt>
<dd><p>Use Normal Distribution Quantile to preprocess a dataset.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be normal distribution quantile.</p></li>
<li><p><strong>quantiles</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of quantiles to be computed., defaults to 1000.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.Standard.scale">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.Standard.</span></span><span class="sig-name descname"><span class="pre">scale</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.Standard.scale" title="Permalink to this definition"></a></dt>
<dd><p>Scale a dataset</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be scaled.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.Standard.standardize">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.Standard.</span></span><span class="sig-name descname"><span class="pre">standardize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.Standard.standardize" title="Permalink to this definition"></a></dt>
<dd><p>Standardize a dataset</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be standarized.</p>
</dd>
</dl>
</dd></dl>

<span class="target" id="module-bioscience.preprocess.RnaSeq"></span><dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.RnaSeq.cpm">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.RnaSeq.</span></span><span class="sig-name descname"><span class="pre">cpm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.RnaSeq.cpm" title="Permalink to this definition"></a></dt>
<dd><p>Apply CPM (counts per million) method preprocessing to a dataset</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be preprocess.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.RnaSeq.fpkm">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.RnaSeq.</span></span><span class="sig-name descname"><span class="pre">fpkm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.RnaSeq.fpkm" title="Permalink to this definition"></a></dt>
<dd><p>Apply FPKM (transcript per million mapped reads) method preprocessing to a dataset</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be preprocess.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.RnaSeq.tpm">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.RnaSeq.</span></span><span class="sig-name descname"><span class="pre">tpm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.RnaSeq.tpm" title="Permalink to this definition"></a></dt>
<dd><p>Apply TPM (transcripts-per-million) method preprocessing to a dataset</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be preprocess.</p>
</dd>
</dl>
</dd></dl>

<span class="target" id="module-bioscience.preprocess.Binarization"></span><dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.Binarization.binarize">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.Binarization.</span></span><span class="sig-name descname"><span class="pre">binarize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">soc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.Binarization.binarize" title="Permalink to this definition"></a></dt>
<dd><p>Applying binarisation to a dataset</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be binarized.</p></li>
<li><p><strong>threshold</strong> (<em>float</em><em>, </em><em>optional</em>) – Feature values below or equal to this are replaced by 0, above it by 1. Threshold may not be less than 0 for operations on sparse matrices, defaults to 0.0</p></li>
<li><p><strong>soc</strong> (<em>int</em><em>, </em><em>optional</em>) – Threshold representing the number of ones each row of the dataset should have as a minimum. If a row does not exceed this threshold, it shall be removed from the dataset, defaults to None.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.preprocess.Binarization.binarizeLevels">
<span class="sig-prename descclassname"><span class="pre">bioscience.preprocess.Binarization.</span></span><span class="sig-name descname"><span class="pre">binarizeLevels</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inactiveLevel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">activeLevel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">step</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">soc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.preprocess.Binarization.binarizeLevels" title="Permalink to this definition"></a></dt>
<dd><p>Generate multiple binary datasets by applying fuzzy logic.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object to be binarized.</p></li>
<li><p><strong>inactiveLevel</strong> (<em>float</em><em>, </em><em>optional</em>) – If an element in the dataset is below this threshold it is considered an inactive gene (value equal to 0 in a binarised dataset).</p></li>
<li><p><strong>activeLevel</strong> (<em>float</em><em>, </em><em>optional</em>) – If an item in the dataset is above this threshold it is considered an active gene (value equal to 1 in a binarised dataset).</p></li>
<li><p><strong>cut</strong> (<em>float</em><em>, </em><em>optional</em>) – Value used in fuzzy logic to determine up to which value an active gene is to be considered. This value will assist in the creation of multiple binarised datasets.</p></li>
<li><p><strong>step</strong> (<em>float</em><em>, </em><em>optional</em>) – Value used in fuzzy logic to determine how much the value in fuzzy logic should be lowered for each binary dataset generated. This value will assist in the creation of multiple binarised datasets.</p></li>
<li><p><strong>soc</strong> (<em>int</em><em>, </em><em>optional</em>) – Threshold representing the number of ones each row of the dataset should have as a minimum. If a row does not exceed this threshold, it shall be removed from the dataset, defaults to None.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</section>
<section id="bioscience-datamining">
<h2>bioscience.dataMining<a class="headerlink" href="#bioscience-datamining" title="Permalink to this headline"></a></h2>
<section id="module-bioscience.dataMining.biclustering.Biclustering">
<span id="biclustering"></span><h3>Biclustering<a class="headerlink" href="#module-bioscience.dataMining.biclustering.Biclustering" title="Permalink to this headline"></a></h3>
<dl class="py function">
<dt class="sig sig-object py" id="bioscience.dataMining.biclustering.Biclustering.bibit">
<span class="sig-prename descclassname"><span class="pre">bioscience.dataMining.biclustering.Biclustering.</span></span><span class="sig-name descname"><span class="pre">bibit</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cMnr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cMnc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">deviceCount</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mode</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">debug</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.dataMining.biclustering.Biclustering.bibit" title="Permalink to this definition"></a></dt>
<dd><p>Main function processing the BiBit Biclustering algorithm.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object store the data of input file.</p></li>
<li><p><strong>cMnr</strong> (<em>int</em><em>, </em><em>optional</em>) – Minimum number of rows to build a valid bicluster, defaults to 2.</p></li>
<li><p><strong>cMnc</strong> (<em>int</em><em>, </em><em>optional</em>) – Minimum number of columns to build a valid bicluster, defaults to 2.</p></li>
<li><p><strong>deviceCount</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of GPU devices to execute, defaults to 1.</p></li>
<li><p><strong>mode</strong> (<em>boolean</em><em>, </em><em>optional</em>) – Type of execution of the algorithm: <cite>mode=1</cite> for sequential execution, <cite>mode=2</cite> for parallel execution on CPUs and <cite>mode=3</cite> for execution on a multi-GPU architecture. Defaults to 1.</p></li>
<li><p><strong>mode</strong> – Attribute used to run the algorithm in debug mode, defaults to False</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A set of BiclusteringModel objects that stores all biclusters generated by the BiBit algorithm.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>set(<a class="reference internal" href="#bioscience.base.models.BiclusteringModel" title="bioscience.base.models.BiclusteringModel"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.BiclusteringModel</span></code></a>)</p>
</dd>
</dl>
</dd></dl>

<span class="target" id="module-bioscience.dataMining.biclustering.BiBit"></span><dl class="py function">
<dt class="sig sig-object py" id="bioscience.dataMining.biclustering.BiBit.process">
<span class="sig-prename descclassname"><span class="pre">bioscience.dataMining.biclustering.BiBit.</span></span><span class="sig-name descname"><span class="pre">process</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataset</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cMnr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cMnc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">deviceCount</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mode</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">debug</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.dataMining.biclustering.BiBit.process" title="Permalink to this definition"></a></dt>
<dd><p>Sub-function processing the BiBit Biclustering algorithm.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataset</strong> (<a class="reference internal" href="#bioscience.base.models.Dataset" title="bioscience.base.models.Dataset"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.Dataset</span></code></a>) – The dataset object store the data of input file.</p></li>
<li><p><strong>cMnr</strong> (<em>int</em>) – Minimum number of rows to build a valid bicluster.</p></li>
<li><p><strong>cMnc</strong> (<em>int</em>) – Minimum number of columns to build a valid bicluster.</p></li>
<li><p><strong>deviceCount</strong> (<em>int</em>) – Number of GPU devices to execute</p></li>
<li><p><strong>mode</strong> (<em>boolean</em>) – Type of execution of the algorithm: <cite>mode=1</cite> for sequential execution, <cite>mode=2</cite> for parallel execution on CPUs and <cite>mode=3</cite> for execution on a multi-GPU architecture.</p></li>
<li><p><strong>mode</strong> – Attribute used to run the algorithm in debug mode.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A BiclusteringModel object that stores all biclusters generated by the BiBit algorithm.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#bioscience.base.models.BiclusteringModel" title="bioscience.base.models.BiclusteringModel"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioscience.base.models.BiclusteringModel</span></code></a></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="bioscience.dataMining.biclustering.BiBit.threadsPerDevice_64">
<span class="sig-prename descclassname"><span class="pre">bioscience.dataMining.biclustering.BiBit.</span></span><span class="sig-name descname"><span class="pre">threadsPerDevice_64</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">resultsQueue</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">i</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bicsPerGpuPrevious</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">patternsPerRun</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mInputData</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">m</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cMnr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cMnc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">debug</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#bioscience.dataMining.biclustering.BiBit.threadsPerDevice_64" title="Permalink to this definition"></a></dt>
<dd><p>Function used for the creation of a multi-GPU architecture.</p>
</dd></dl>

</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../userGuide/results.html" class="btn btn-neutral float-left" title="Results" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../contributors/contributing.html" class="btn btn-neutral float-right" title="Contributing to bioScience" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, bioScience Core Development Team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>